src/tools/lint-docs/src/lib.rs

```diff
@@ -441,10 +441,14 @@ impl<'a> LintExtractor<'a> {
         fs::write(&tempfile, source)
             .map_err(|e| format!("failed to write {}: {}", tempfile.display(), e))?;
         let mut cmd = Command::new(self.rustc_path);
-        if options.contains(&"edition2015") {
-            cmd.arg("--edition=2015");
-        } else {
+        if options.contains(&"edition2024") {
+            cmd.arg("--edition=2024");
+        } else if options.contains(&"edition2021") {
+            cmd.arg("--edition=2021");
+        } else if options.contains(&"edition2018") {
             cmd.arg("--edition=2018");
+        } else {
+            cmd.arg("--edition=2015");
```

Comment:
nit: we could `panic!` when we detect an unknown edition, so that future users are not surprised by the weird behavior
```suggestion
        } else if options.contains(&"edition") {
            panic!("lint-docs: unknown edition");
        } else {
            cmd.arg("--edition=2015");
```